### PR TITLE
Optimize tokenizer token memoization

### PR DESCRIPTION
**Pull request by pmaupin**
Pull separate memoization function into main function and optimize code paths.

Increases overall test performance by around 2% on my machine.

### CODE REVIEW

File: pdfrw/tokens.py [hunk 1]
```diff
@@ -95,24 +83,23 @@ def _gettoks(self, startloc, cacheobj=_cacheobj,
         fdata = self.fdata
         current = self.current = [(startloc, startloc)]
         cache = {}
+        get_cache = cache.get
         while 1:
             for match in findtok(fdata, current[0][1]):
                 current[0] = tokspan = match.span()
                 token = match.group(1)
                 firstch = token[0]
+                toktype = intern
```
Comment (takluyver) [inline pdfrw/tokens.py]: It seems a bit weird to me that `toktype` can be the intern function, but I'll leave that up to you. I know you're more willing to trade some readability for performance than I would be. ;-)

**Comment by takluyver** [comment]
Other than that, this looks fine to me.

File: pdfrw/tokens.py (hunk 1 above)
Comment (pmaupin) [inline pdfrw/tokens.py]: What can I say?  C is the new assembly; Python is the new C.  And I don't even drop down to the new assembly unless absolutely required :-)

Thanks for looking it over.

Comment (takluyver) [inline pdfrw/tokens.py]: I wonder what's the new Python? :-P

**Pull request merged**